tests/unit/test-util/test.file-filter.js

```diff
@@ -74,12 +74,22 @@ describe('util/file-filter', () => {
 
     it('ignore artifactsDir and its content', () => {
       const filter = new FileFilter({
+        sourceDir: '.',
         artifactsDir: 'artifacts',
       });
       assert.equal(filter.wantFile('artifacts'), false);
       assert.equal(filter.wantFile('artifacts/some.js'), false);
     });
 
+    it('does not ignore an artifactsDir outside of sourceDir', () => {
+      const filter = new FileFilter({
+        artifactsDir: '.',
+        sourceDir: 'dist',
+      });
+      assert.equal(filter.wantFile('file'), true);
```

Comment:
Here's the detail. Currently,

1. All ignored patterns are resolved with `sourceDir`.
2. Resolve other paths (e.g. `sourceDir`, `artifactsDir`, `path` in `wantFile`) with `cwd`.
3. Since all ignored patterns are resolved with `sourceDir`, everything outside of `sourceDir` won't be ignored.

```JavaScript
const fileFilter = new FileFilter({
	sourceDir: 'dist/',
	ignoreFiles: ['some-script.js']
});
assert.equal(fileFilter.wantFile('dist/some-script.js'), false); // ignored by ignoreFiles
assert.equal(fileFilter.wantFile('some-script.js'), true);	// outside of sourceDir
```

I'd like to discuss some problems here:

**Is resolving ignored patterns with `sourceDir` make sense?** Or should we resolve ignored patterns with `cwd`? Which would look like:
```diff
const fileFilter = new FileFilter({
	sourceDir: 'dist/',
	ignoreFiles: ['some-script.js']
});
-assert.equal(fileFilter.wantFile('dist/some-script.js'), false); // ignored by ignoreFiles
+assert.equal(fileFilter.wantFile('dist/some-script.js'), true); // some-script.js is resolved with cwd instead of dist
-assert.equal(fileFilter.wantFile('some-script.js'), true);	// outside of sourceDir
+assert.equal(fileFilter.wantFile('some-script.js'), false);	// ignored by ignoreFiles
```

Second, **since we make `sourceDir` mandatory now, should we ignore all files outside of `sourceDir`?**
```diff
const fileFilter = new FileFilter({
	sourceDir: 'dist/',
	ignoreFiles: ['some-script.js']
});
assert.equal(fileFilter.wantFile('dist/some-script.js'), false); // ignored by ignoreFiles
-assert.equal(fileFilter.wantFile('some-script.js'), true);	// outside of sourceDir
+assert.equal(fileFilter.wantFile('some-script.js'), false);	// outside of sourceDir
```